server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -266,7 +245,7 @@ private MessageWithId handleOutboxMessages(CreationMessageEntry entry, MailboxSe
     
     private boolean isAppendToMailboxWithRole(Role role, CreationMessage entry, MailboxSession mailboxSession) throws MailboxException {
         return getMailboxWithRole(mailboxSession, role)
-                .map(entry::isIn)
+                .map(entry::isInOnly)
```

Comment:
it looks like it's not what is described in commit title

Reply:
Yes this one deserves to be in 

```
JAMES-2214 Remove redundant OUTBOX checking 
```

I'll manage that on next rebase.